plugins/package-managers/cocoapods/src/main/kotlin/CocoaPods.kt

```diff
@@ -236,7 +236,13 @@ private fun getPackageReferences(podfileLock: File): Set<PackageReference> {
 
     fun parseNameAndVersion(entry: String): Pair<String, String?> {
         val info = entry.split(' ', limit = 2)
-        return info[0] to info.getOrNull(1)?.removeSurrounding("(", ")")
+        val name = info[0]
+
+        // A version entry could look something like "(6.3.0)", "(= 2021.06.28.00-v2)", "(~> 8.15.0)", etc. Also see
+        // https://guides.cocoapods.org/syntax/podfile.html#pod.
+        val version = info.getOrNull(1)?.removeSurrounding("(", ")")?.substringAfterLast(' ')
+
+        return name to version
```

Comment:
Can it now happen that an entry in `versionForName` already exists and get overwritten in line 257?
If so, would the value always be overwritten by its identity, or could the value change?

Reply:
CocoaPods does not support having the same dependency in different versions, so *if* the version would be specified twice (which I've never seen in a `Podfile.lock`), it absolutely should be the same, and a value would get overwritten by its identity.